File: Leetcode/LinkedList/Swap Nodes in Pairs.py
#   O(n)   O(1)
class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        dummy=ListNode(0)
        dummy.next=head
        prev=dummy
        curr=head
        while(prev.next and prev.next.next):
            temp = curr.next.next
            prev.next=curr.next
            curr.next.next=curr
            curr.next=temp
            prev=curr
            curr=curr.next
        return dummy.next
 

File: finalFacebook/Swap Nodes in Pairs.java
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode prev = dummy;
        ListNode first = head;
        while(first != null && first.next != null) {
            ListNode second = first.next;
            first.next = second.next;
            second.next = first;
            prev.next = second;
            prev = prev.next.next;
            first = prev.next; 
        }
        return dummy.next;
    }
}